src/datasets/arrow_dataset.py

```diff
@@ -3371,71 +3371,100 @@ def reduce(
+             input_columns (`Optional[Union[str, List[str]]]`, defaults to `None`):
+                 The columns to be passed into `function`
+                 as positional arguments. If `None`, a `dict` mapping to all formatted columns is passed as one argument.
+             initializer (`Optional[Any]`, defaults to `None`):
+                 The initial value of the accumulator, it is the first argument of the first application of the function, and hence must be of the same type as the examples.
+                 If `batched=True`, the initializer must be set.
+             batched (`bool`, defaults to `False`):
+                 Provide batch of examples to `function`.
+             batch_size (`int`, *optional*, defaults to `1000`):
+                 Number of examples per batch provided to `function` if `batched=True`.
+                 If `batch_size <= 0` or `batch_size == None`, provide the full dataset as a single batch to `function`.
+             drop_last_batch (`bool`, defaults to `False`):
+                 Whether a last batch smaller than the batch_size should be
+                 dropped instead of being processed by the function.
+             keep_in_memory (`bool`, defaults to `False`):
+                 Keep the dataset in memory instead of writing it to a cache file.
+             load_from_cache_file (`bool`, defaults to `True` if caching is enabled):
+                 If a cache file storing the current computation from `function`
+                 can be identified, use it instead of recomputing.
+             cache_file_name (`str`, *optional*, defaults to `None`):
+                 Provide the name of a path for the cache file. It is used to store the
+                 results of the computation instead of the automatically generated cache file name.
+             fn_kwargs (`Dict`, *optional*, defaults to `None`):
+                 Keyword arguments to be passed to `function`.
+             num_proc (`int`, *optional*, defaults to `None`):
+                 Max number of processes when generating cache. Already cached shards are loaded sequentially.
+             new_fingerprint (`str`, *optional*, defaults to `None`):
+                 The new fingerprint of the dataset after transform.
+                 If `None`, the new fingerprint is computed using a hash of the previous fingerprint, and the transform arguments.
+             desc (`str`, *optional*, defaults to `None`):
+                 Meaningful description to be displayed alongside with the progress bar while mapping examples.
+
+         Example:
+
+         ```py
+         >>> from datasets import load_dataset
+         >>> string_ds = load_dataset("rotten_tomatoes", split="validation")
+         >>> str_concat = lambda accumulator, example: accumulator + ' ' + example
+         >>> result = string_ds[0:3].reduce(str_concat, input_columns="text")
```

Comment:
Usually reduce operations are used to compute aggregates, not merging examples together (you could just do `" ".join(ds["text"])` here)

Maybe we can find a use case that only `reduce` can achieve ? Maybe computing the `sum` of an integer column.
An actual use case would be to sum all the review of the `imdb` dataset, which can be used to derive the average review score.